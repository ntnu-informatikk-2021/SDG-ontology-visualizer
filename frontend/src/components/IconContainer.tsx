import React, { useState, useEffect } from 'react';
import { Box, Container } from '@chakra-ui/react';
import { getSustainabilityGoals } from '../api/ontologies';
import {  Node, SustainabilityGoal } from '../types/ontologyTypes';

// type IconContainerProps = {
//   sustainabilityNode: Node;
// };

const IconContainer = () => {
    const [sustainabilityGoals, setSustainabilityGoals] = useState<SustainabilityGoal>({
        instancesOf: '',
        label: '',
        icon: '',
      });
    
      const loadSustainabilityGoals = async () => {
        const data = await getSustainabilityGoals(node.id);
        setSustainabilityGoals(data);
      };
    
      useEffect(() => {
        loadSustainabilityGoals();
      }, []);

       const [sustainabilityGoals, setSustainabilityGoals] = useState<SustainabilityGoal>({
        instancesOf: '',
        label: '',
        icon: '',
      });
    
      const loadSustainabilityGoals = async () => {
        const data = await getSustainabilityGoals(imageid.id);
        setSustainabilityGoals(data);
      };
    
      useEffect(() => {
        loadSustainabilityGoals();
      }, []);
  
      return(
  <Container paddingTop={20} centerContent>
    <Box
      borderWidth="1g"
      borderRadius={5}
      p={0}
      w="150%"
      overflow="hidden"
      border="2px"
      borderColor="linkedin.400"
    >
      <p>{sustainabilityGoals.label}</p>
      <img src={sustainabilityGoals{imageid}.icon} alt="fn.no" />
    </Box>
  </Container>
      )};

      export default IconContainer;