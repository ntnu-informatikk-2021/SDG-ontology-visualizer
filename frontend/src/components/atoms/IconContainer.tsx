import React from 'react';
import { Image } from '@chakra-ui/react';
import { SustainabilityGoal } from '../../types/ontologyTypes';
import { MotionBox } from '../../types/react/componentTypes';

type IconContainerProps = {
  sustainabilityNode: SustainabilityGoal;
  onClick: (sdg: SustainabilityGoal) => void;
};

const IconContainer: React.FC<IconContainerProps> = ({
  sustainabilityNode,
  onClick,
}: IconContainerProps) => (
  <MotionBox
    p={0}
    whileHover={{ scale: 1.05 }}
    _hover={{
      cursor: 'pointer',
    }}
    onClick={() => onClick(sustainabilityNode)}
  >
    <Image
      src={sustainabilityNode.icon}
      borderRadius="lg"
      overflow="hidden"
      alt={sustainabilityNode.label}
      boxSize="250"
      object-fit="cover"
    />
  </MotionBox>
);

export default IconContainer;
