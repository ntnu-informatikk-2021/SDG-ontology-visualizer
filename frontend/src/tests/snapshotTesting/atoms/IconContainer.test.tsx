import React from 'react';
import renderer from 'react-test-renderer';

import IconContainer from '../../../components/atoms/IconContainer';

const goal1 = {
  instancesOf: 't2',
  label: 'hei',
  icon: 'golum',
};

it('renders when there are no items', () => {
  const tree = renderer
    .create(<IconContainer sustainabilityNode={goal1} onClick={() => {}} />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});
