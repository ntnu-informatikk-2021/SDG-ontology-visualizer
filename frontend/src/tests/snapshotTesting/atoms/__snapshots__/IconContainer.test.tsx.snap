// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there are no items 1`] = `
<div
  className="css-1sjj4m4"
  onClick={[Function]}
  style={Object {}}
>
  <img
    alt="hei"
    className="chakra-image__placeholder css-1v14er2"
    object-fit="cover"
  />
</div>
`;
